test/serialization/config/CustomSerializerTest.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var Controller = require('../../RC');
+var expect = require('chai').expect;
+
+var Client = require('../../../').Client;
+var Musician = require('./Musician').Musician;
+
+describe('CustomSerializer', function () {
+    var cluster;
+
+    before(function () {
+        process.chdir('test/serialization/config');
+        return Controller.createCluster(null, null).then(function (cl) {
+            cluster = cl;
+            return Controller.startMember(cluster.id);
+        });
+    });
+
+    after(function () {
+        Controller.shutdownCluster(cluster.id);
+        process.chdir('../../../');
+    })
+
+    it('should be configured declaratively', function () {
```

Comment:
This test does not verify custom serializer is used. The system can fallback to Json serializer and the test would still pass. Moreover I think this is what happens right now. Since `Musician` does not have `hzGetCustomId` method, it cannot match the custom serializer to musicion object.